test/matrixnormal.jl

```diff
@@ -0,0 +1,128 @@
+using Distributions, Random
+using Test, LinearAlgebra, PDMats
+
+#  dimensions
+
+n = 3
+p = 6
+
+#  parameters
+
+M    = randn(n, p)
+U    = rand(InverseWishart(n + 2, Matrix(1.0I, n, n)))
+V    = rand(InverseWishart(p + 2, Matrix(1.0I, p, p)))
+UF32 = Matrix{Float32}(U)
+VBF  = Matrix{BigFloat}(V)
+PDU  = PDMat(U)
+PDV  = PDMat(V)
+m    = randn(n)
+w    = randn(p)
+μ    = randn()
+σ    = 2.0rand()
+σ²   = σ ^ 2
+
+#  distribution instances
+
+D = MatrixNormal(M, U, V)  #  n x p
+G = MatrixNormal(p, n)     #  p x n
+L = MatrixNormal(reshape(m, n, 1), U, reshape([σ²], 1, 1))  #  n x 1
+H = MatrixNormal(reshape(w, 1, p), reshape([σ²], 1, 1), V)  #  1 x p
+K = MatrixNormal(reshape([μ], 1, 1), reshape([σ], 1, 1), reshape([σ], 1, 1))  #  1 x 1
+
+d = vec(D) # MvNormal(vec(M), V ⊗ U)
+g = MvNormal( Matrix(1.0I, p*n, p*n) )
+l = MvNormal(m, σ² * U)
+h = MvNormal(w, σ² * V)
+k = Normal(μ, σ)
+
+#  draws
+
+A = rand(D)
+B = rand(G)
+X = rand(L)
+Y = rand(H)
+Z = rand(K)
+
+a = vec(A)
+b = vec(B)
+x = vec(X)
+y = vec(Y)
+z = Z[1, 1]
+
+#  Call every constructor to catch errors
+
+MatrixNormal(M, U,        PDV.chol)
+MatrixNormal(M, U,        PDV)
+MatrixNormal(M, PDU.chol, V)
+MatrixNormal(M, PDU.chol, PDV.chol)
+MatrixNormal(M, PDU.chol, PDV)
+MatrixNormal(M, PDU,      V)
+MatrixNormal(M, PDU,      PDV.chol)
+MatrixNormal(M, PDU,      PDV)
```

Comment:
Should these be something like `@test MatrixNormal(M, U, V) isa MatrixNormal` just for the sake of these being test failures rather than errors outside of a testset?

Also should this whole file be wrapped in a `@testset "MatrixNormal begin ... end`?

Reply:
Great, thanks. I will do this shortly.